control-plane/pkg/reconciler/broker/namespaced_broker.go

```diff
@@ -112,9 +117,56 @@ func (r *NamespacedReconciler) FinalizeKind(ctx context.Context, broker *eventin
 
 	r.IPsLister.Unregister(types.NamespacedName{Namespace: broker.Namespace, Name: broker.Name})
 
+	err := r.cleanUpClusterScopedResources(ctx, broker)
+	if err != nil {
+		// we don't let this error propagate, as we don't want to requeue this forever
+		logging.FromContext(ctx).Errorw("Error while cleaning up cluster scoped resources for KafkaNamespaced brokers in the namespace", zap.Error(err))
+		return result
+	}
+
 	return result
 }
 
+// cleanUpClusterScopedResources deletes the cluster scoped resources created for the data plane in the namespace, if
+// this is the last namespaced broker in the namespace
+func (r *NamespacedReconciler) cleanUpClusterScopedResources(ctx context.Context, broker *eventing.Broker) error {
+	var list []*eventing.Broker
+	var err error
+
+	// TODO: can we use a label to select KafkaNamespaced brokers instead of listing all brokers in the namespace?
```

Comment:
// TODO: can we use a label to select KafkaNamespaced brokers instead of listing all brokers in the namespace?

We currently use annotations for the class. We can use a separate label that can be added to namespaced brokers. I now think it is probably not worth the effort though